vendor/github.com/ethereum/go-ethereum/whisper/whisperv5/doc.go

```diff
@@ -99,3 +100,34 @@ type NotificationServer interface {
 	// Stop stops notification sending loop, releasing related resources
 	Stop() error
 }
+
+// RPCMessageState holds the current delivery state of a whisper rpc message.
+type RPCMessageState struct {
+	Reason    error             `json:"reason"`
+	Envelope  Envelope          `json:"envelope"`
+	Timestamp time.Time         `json:"timestamp"`
+	Source    NewMessage        `josn:"source"`
+	Status    message.Status    `json:"status"`
+	Direction message.Direction `json:"direction"`
+	Received  *ReceivedMessage  `json:"received"`
+}
+
+// P2PMessageState holds the current delivery status of a whisper p2p message.
+type P2PMessageState struct {
```

Comment:
Question - types `P2PMessageState` and `RPCMessageState` are equal. Why they're implemented as two differents structs? Would not `MessageState` be enough?